### PR TITLE
Update Bonsai readme

### PR DESCRIPTION
**Pull request by rlukata**
This PR updates the README and some docs. It also changes the CI workflow to `doc-test` all crates in bonsai excluding `bonsai-ethereum-contracts`.

### CODE REVIEW

**Comment by flaub** [comment]
Why didn't CI catch this?

**Comment by rlukata** [comment]
> Why didn't CI catch this?

I think it's because we had the following code in the Cargo.toml:
```
[lib]
# Disables the doctest from the README because it imports external components and make requests
doctest = false
```

**Comment by rlukata** [comment]
I may have opened the gates of hell by running doc-tests for all crates within `bonsai`. I'm going to take a stab at fixing them, but let me know if there's a different approach you'd like me to explore @flaub.

**Comment by rlukata** [comment]
The `bonsai-ethereum-contracts` crate in the `ethereum` directory was complaining about some issues with the doc-tests. Given that the code is expanded by a macro that we do not maintain, I figured it makes sense to simply ignore the doc-tests for that crate as we already test the crate.

File: bonsai/sdk/README.md [hunk 1]
```diff
@@ -4,27 +4,31 @@ A library to handle HTTP REST requests to the Bonsai-alpha prover interface
 
 ## Example Usage
 
-```rust
+```rust,no_run
```
Comment (rlukata) [inline bonsai/sdk/README.md:7]: `no_run` means the code will compile but won't run

File: bonsai/sdk/README.md [hunk 2]
```diff
@@ -4,27 +4,31 @@ A library to handle HTTP REST requests to the Bonsai-alpha prover interface
...
+use std::time::Duration;
+
 use anyhow::Result;
 use bonsai_sdk::alpha as bonsai_sdk;
-use methods::{METHOD_NAME_ELF, METHOD_NAME_ID};
-use risc0_zkvm::{serde::to_vec, Receipt};
-use std::time::Duration;
+use hello_world_methods::{MULTIPLY_ELF, MULTIPLY_ID};
```
Comment (rlukata) [inline bonsai/sdk/README.md:12]: I'd be happy changing this to use other methods, let me know.

**Review by mothran — APPROVED** [review]
This looks good on my end, great that we got the the code samples building again!

**Review by flaub — APPROVED** [review]
Amazing!

**Pull request merged**